### PR TITLE
CCIP-4320 : Part 1 - Parameterize ocr inputs 

### PR DESCRIPTION
**Pull request by AnieeG**
- Enables parameterization of OCR inputs
- Adds AddLanesWithTestRouter changeset  - This is useful to test the whole deployment and on successful send with test router, the lanes can be enabled with main router ( not included in this PR)

### CODE REVIEW

**Review by connorwstein — DISMISSED** [review]

File: deployment/ccip/changeset/add_lane.go [hunk 1]
```diff
@@ -15,25 +17,103 @@ import (
...
+
 type InitialPrices struct {
 	LinkPrice *big.Int // USD to the power of 18 (e18) per LINK
 	WethPrice *big.Int // USD to the power of 18 (e18) per WETH
 	GasPrice  *big.Int // uint224 packed gas price in USD (112 for exec // 112 for da)
 }
 
+type Lane struct {
+	From uint64
+	To   uint64
```
Comment (makramkd) [inline deployment/ccip/changeset/add_lane.go]: nit: Lets make an effort to standardize the source/dest jargon where possible to reduce this kinda unneeded naming variance

```suggestion
	SourceSelector uint64
	DestSelector   uint64
```

File: deployment/ccip/changeset/add_lane.go [hunk 2]
```diff
@@ -15,25 +17,103 @@ import (
...
+}
+
+type AddLanesConfig struct {
+	FeeQuoterDestChainConfigArgs map[uint64]map[uint64]fee_quoter.FeeQuoterDestChainConfig
+	InitialPricesByChain         map[uint64]InitialPrices
+	ChainPairs                   []Lane
```
Comment (makramkd) [inline deployment/ccip/changeset/add_lane.go]:
```suggestion
	Lanes                   []Lane
```

Seems like a natural name for a slice of lane objects

File: deployment/ccip/changeset/add_lane.go [hunk 3]
```diff
@@ -15,25 +17,103 @@ import (
...
+	}
+	return nil
+}
+
+func AddLaneWithDefaultPrices(e deployment.Environment, state CCIPOnChainState, from, to uint64, isTestRouter bool) error {
+	return AddLane(e, state, from, to, DefaultInitialPrices, isTestRouter, DefaultFeeQuoterDestChainConfig())
 }
 
-func AddLane(e deployment.Environment, state CCIPOnChainState, from, to uint64, initialPrices InitialPrices) error {
+func AddLane(e deployment.Environment, state CCIPOnChainState, from, to uint64, initialPrices InitialPrices, isTestRouter bool, feeQuoterDestChainConfig fee_quoter.FeeQuoterDestChainConfig) error {
```
Comment (makramkd) [inline deployment/ccip/changeset/add_lane.go]: Would be nice if we created data structures that are used across all of these `AddLane` variants. At a high level, if you want to Add a single Lane, you'd do AddLane(..., laneConfig). If you want to add many lanes, you'd need to pass in an array of lane configs, `AddLanes(..., []LaneConfig{...})`.

Comment (makramkd) [reply deployment/ccip/changeset/add_lane.go]: Or do you really need a `AddLane` and `AddLanes` to have separate implementations? IMO either have one (`AddLanes` - can be called with a single lane config, adds a single lane) or the other (`AddLane` - can be called in a loop for each lane to be added)

File: deployment/ccip/changeset/internal/deploy_home_chain.go [hunk 4]
```diff
@@ -445,25 +443,25 @@ func BuildOCR3ConfigForCCIPHome(
...
-				RMNEnabled:                         os.Getenv("ENABLE_RMN") == "true", // only enabled in manual test
+				RemoteGasPriceBatchWriteFrequency:  commitOffchainCfg.RemoteGasPriceBatchWriteFrequency,
+				TokenPriceBatchWriteFrequency:      commitOffchainCfg.TokenPriceBatchWriteFrequency,
+				PriceFeedChainSelector:             commitOffchainCfg.PriceFeedChainSelector,
+				TokenInfo:                          commitOffchainCfg.TokenInfo,
+				NewMsgScanBatchSize:                commitOffchainCfg.NewMsgScanBatchSize,
+				MaxReportTransmissionCheckAttempts: commitOffchainCfg.MaxReportTransmissionCheckAttempts,
+				MaxMerkleTreeSize:                  commitOffchainCfg.MaxMerkleTreeSize,
+				SignObservationPrefix:              commitOffchainCfg.SignObservationPrefix,
+				RMNEnabled:                         commitOffchainCfg.RMNEnabled, // only enabled in manual test
```
Comment (makramkd) [inline deployment/ccip/changeset/internal/deploy_home_chain.go]: Comment no longer relevant right?

File: deployment/common/types/types.go [hunk 5]
```diff
@@ -23,3 +25,58 @@ type MCMSWithTimelockConfig struct {
...
+	DeltaCertifiedCommitRequest             time.Duration
+	DeltaStage                              time.Duration
+	Rmax                                    uint64
+	MaxDurationQuery                        time.Duration
+	MaxDurationObservation                  time.Duration
+	MaxDurationShouldAcceptAttestedReport   time.Duration
+	MaxDurationShouldTransmitAcceptedReport time.Duration
+}
+
+func (params OCRParameters) Validate() error {
```
Comment (makramkd) [inline deployment/common/types/types.go:44]: libocr already has exhaustive validation for this, can we call out to that instead of doing this here?

Comment (AnieeG) [reply deployment/common/types/types.go:44]: I created a ticket to handle this - https://smartcontract-it.atlassian.net/browse/CCIP-4332

**Review by connorwstein — APPROVED** [review]

**Review by b-gopalswami — APPROVED** [review]

**Pull request merged**